### DISCUSSION TITLE
Restart Laravel Octane after deploy without supervisor permissions

### DISCUSSION THREAD

**Discussion by eusonlito** [laravel/framework#42431] [Q&A]
Hello! I have a supervisor task to run Laravel Octane:

```
[program:laravel-octane-worker]
process_name=%(program_name)s_%(process_num)02d
command=php8.1 /var/www/artisan octane:start --server=roadrunner --max-requests=1000 --workers=2 --task-workers=2 --port=8081
autostart=true
autorestart=true
user=www-data
redirect_stderr=true
stdout_logfile=/var/www/storage/logs/supervisor-octane-worker.log
stopwaitsecs=3600
```

But if I try to manually restart octane from command line after deploy (I haven't permission to restart supervisor), I get the error:

```
$ php artisan octane:reload

   INFO  Reloading workers...

In ServerProcessInspector.php line 56:
                                                                          
  Cannot reload RoadRunner: 2022/05/18 22:36:29 resetting plugin: [http]  
```

Then, who can I restart the octane service after deploy?

Thanks!

**Comment by tomvo** [comment]
@eusonlito restart the supervisor process using `supervisorctl` so, `supervisorctl restart laravel-octane-worker`

**Comment by Kamandlou** [comment]
Restart the supervisor process using `supervisorctl restart all`

**Comment by grantholle** [comment]
Hi @eusonlito, did you ever figure this out? I am in the same situation, except it doesn't throw an error when doing `octane:reload`. I also would not like to give the user permission to restart supervisor. If I run `octane:status`, it does say that the server is running, but `octane:reload` does not restart the process

**Comment by leninsheikh** [comment]
@eusonlito 

### ✅ How to reload octane server when its running under supervisor?
**there is two way you can do this-**

1.  By stopping  worker process. Following artisan command will stop the worker process, then supervisor will auto restart the prcoess (assuming you configure supervisor to auto restart). This way you do not need sudo password to restart worker.

    `php artisan octane:stop`



3.  By restarting the supervisor, you will need sudo password, which you may not want.

    `sudo  supervisorctl restart all` #will restart all superviros process

      or

    `sudo supervisorctl restart  laravel-octane-worker:*`  #only restart specific supervisor program, dont forget to change **laravel-octane-worker** with you program name.
